plugins/ingest/src/main/java/org/elasticsearch/ingest/processor/ConfigurationUtils.java

```diff
@@ -149,4 +149,15 @@ private static String readString(String propertyName, Object value) {
             throw new IllegalArgumentException("property [" + propertyName + "] isn't a map, but of type [" + value.getClass().getName() + "]");
         }
     }
+
+    /**
+     * Returns and removes the specified property as an {@link Object} from the specified configuration map.
+     */
+    public static Object readObject(Map<String, Object> configuration, String propertyName) {
+        if (configuration.containsKey(propertyName)) {
```

Comment:
when we have `field: null` containsKey will return true, but the method will return null as remove returns null, which was the actual value. I think that null shouldn't be returned and we should throw an exception instead, like we do in other methods. Just call remove and check the returned value, without containsKey.